mbedtls/src/cipher/raw/mod.rs

```diff
@@ -392,26 +426,26 @@ impl Cipher {
     pub fn encrypt_auth_inplace(
         &mut self,
         ad: &[u8],
-        data: &mut [u8],
-        tag: &mut [u8],
+        data_with_tag: &mut [u8],
+        tag_len: usize,
     ) -> Result<usize> {
 
-        let iv = self.inner.iv;
-        let iv_len = self.inner.iv_size;
-        let mut olen = data.len();
+        let iv = self.inner.private_iv;
+        let iv_len = self.inner.private_iv_size;
+        let mut olen = data_with_tag.len();
         unsafe {
-            cipher_auth_encrypt(
+            cipher_auth_encrypt_ext(
                 &mut self.inner,
                 iv.as_ptr(),
                 iv_len,
                 ad.as_ptr(),
                 ad.len(),
-                data.as_ptr(),
-                data.len(),
-                data.as_mut_ptr(),
+                data_with_tag.as_ptr(),
+                data_with_tag.len() - tag_len,
```

Comment:
Since `unsafe`, do you know what is the behavior if this value is `<= 0`? I would assume `CipherBadInputData`.

Reply:
Got it. Will update all similar value check too.

Reply:
Updated